### DISCUSSION TITLE
Limiting line to 1 line

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1095] [Questions & Help]
Hi, can i limit the Polyline to only draw 1 line ( basically ending the drawing on the 2nd click and not allowing it to continue for another line

**Comment by Falke-Design** [comment]
```
map.on('pm:drawstart', ({ workingLayer }) => {
  workingLayer.on('pm:vertexadded', (e) => {
    if(e.shape == 'Line' && workingLayer.getLatLngs().length >= 2){
  		map.pm.Draw.Line._finishShape();
  	}
  });
});
```

**Reply by iainxt** [reply]
This works for the initial drawing but does not prevent a user from adding extra lines during edit mode.

![image](https://user-images.githubusercontent.com/146976/165212292-d04c63f8-23c7-4ad4-b497-b94160a77f9c.png)

I am doing a measuring tape like feature where I want a single segment line only.

**Reply by iainxt** [reply]
I see I can disable the middle markers, so that part is ok

        map.pm.enableDraw('Line', {
            hideMiddleMarkers: true
        });
